src/concrete/ml/quantization/post_training.py

```diff
@@ -23,6 +23,99 @@
 from .quantized_ops import QuantizedBrevitasQuant
 from .quantizers import QuantizationOptions, QuantizedArray, UniformQuantizer
 
+# pylint: disable=too-many-lines
+
+
+def _inspect_tree_n_bits(n_bits):
+    """Validate the 'n_bits' parameter for tree-based models.
+
+    This function checks whether 'n_bits' is a valid integer or dictionary.
+    - If 'n_bits' is an integer, it must be a non-null positive, its value is assigned to
+        'op_inputs' and 'op_leaves' bits
+    - If it is a dictionary, it should contain integer values for keys 'op_leaves' and 'op_inputs',
+        where 'op_leaves' should not exceed 'op_inputs'.
```

Comment:
Not sure about the namings here. What about `n_bits_input` `n_bits_output`?

Reply:
NNs have `op_inputs` `op_outputs`. It keeps that style, I think it's fine like this.

Reply:
@jfrery the reason is that QNNs have `model_inputs`, `model_outputs` , `op_inputs` and `op_weights`, so you need to differentiate both. As Andrei said, here, it's taking the same convention

although I agree for trees it's getting q bit misleading without much context but we don't have much choice as we should keep our namings coherent with one another

my main comment on this naming was about `op_leave` (https://github.com/zama-ai/concrete-ml/pull/449#discussion_r1453237293), this introduces a new name into our convention. Although I get why it's more relevant to tree models, I felt keeping `op_weights` was fine imo

Reply:
Well I find op_ not ideal for tree based models. Also if we want to keep consistency then it should be model_inputs instead of op_inputs.

I am really not a fan of having op_leave for many reasons. Most users won't know what leave refer to. Same for op.

The why not `model_input` and `model_output`? It makes sense for the input but would require a change for `model_output`. This will help us maintain the output at a given precision which is not easy with op_leave since this targets every tree instead of the final combination.

Reply:
I agree with you I'm not a fan of `op_leave`. but about having `model_input` and `model_output` only, how are you going to handle the n_bits to use for  quantizing the leaves ? you'll either need an additional parameter similar to the suggested `op_leave` or a way to infer its value based on `model_output` but not sure how this could easily/efficiently be done 🤔

Reply:
@zama-ai/machine-learning, please let's agree on a proper naming.
I suggest: tree_input_bits/tree_leaf_bits or quant_input_bits/quant_leaf_bits

-> model_ouput why not, but doesn't ring like it's the leaves that we are quantizing
-> op_weight, really not a big fan, we are quantizing leaves not weights.

Reply:
I think it's best if you ask directly in the slack channel !